tests/test_diffraction_objects.py

```diff
@@ -223,6 +223,114 @@ def test_on_xtype_bad():
         test.on_xtype("invalid")
 
 
+params_scale_to = [
+    # UC1: same x-array and y-array, check offset
+    (
+        [
+            np.array([10, 15, 25, 30, 60, 140]),
+            np.array([2, 3, 4, 5, 6, 7]),
+            "tth",
+            2 * np.pi,
+            np.array([10, 15, 25, 30, 60, 140]),
+            np.array([2, 3, 4, 5, 6, 7]),
+            "tth",
+            2 * np.pi,
+            None,
+            60,
+            None,
+            2.1,
+        ],
+        ["tth", np.array([4.1, 5.1, 6.1, 7.1, 8.1, 9.1])],
+    ),
+    # UC2: same length x-arrays with exact x-value match
+    (
+        [
+            np.array([10, 15, 25, 30, 60, 140]),
+            np.array([10, 20, 25, 30, 60, 100]),
+            "tth",
+            2 * np.pi,
+            np.array([10, 20, 25, 30, 60, 140]),
+            np.array([2, 3, 4, 5, 6, 7]),
+            "tth",
+            2 * np.pi,
+            None,
+            60,
+            None,
+            0,
+        ],
+        ["tth", np.array([1, 2, 2.5, 3, 6, 10])],
+    ),
+    # UC3: same length x-arrays with approximate x-value match
+    (
+        [
+            np.array([0.12, 0.24, 0.31, 0.4]),
+            np.array([10, 20, 40, 60]),
+            "q",
+            2 * np.pi,
+            np.array([0.14, 0.24, 0.31, 0.4]),
+            np.array([1, 3, 4, 5]),
+            "q",
+            2 * np.pi,
+            0.1,
+            None,
+            None,
+            0,
+        ],
+        ["q", np.array([1, 2, 4, 6])],
+    ),
+    # UC4: different x-array lengths with approximate x-value match
+    (
```

Comment:
A test example for scaling DOs with different array lengths. Here I think it makes more sense to scale them on q=61 (for self) & q=62 (for target).